app/controllers/internships_controller.rb

```diff
@@ -0,0 +1,78 @@
+class InternshipsController < ApplicationController
+  def index
+    @cohort = Cohort.find(params[:cohort_id])
+    @internships = @cohort.internships
+    if current_student
+      @internships = @internships.sort_by { |internship| by_student_interest(internship) }
+    end
```

Comment:
refactor lines 5-7 to just be `@internships = @cohort.internships_sorted_by_interest(current_student)`. put all of the logic of `by_student_interest` and the above `sort_by` into the `Cohort#internships_sorted_by_interest` method. Write unit tests around it.